financialaid/views_test.py

```diff
@@ -269,3 +245,167 @@ def test_review_financial_aid_view_with_invalid_filter_and_sorting(self):
+        assert called_kwargs['body'] == FINANCIAL_AID_APPROVAL_MESSAGE_BODY
+        assert called_kwargs['recipient'] == self.profile.user.email
+        # Application is approved for a different tier program
+        financialaid.status = FinancialAidStatus.PENDING_MANUAL_APPROVAL
+        financialaid.save()
+        self.financial_review_data["tier_program_id"] = self.tier_programs["50k"].id
+        resp = self.client.post(self.action_url, data=self.financial_review_data)
+        # Re-retrieve financialaid object
+        financialaid = FinancialAid.objects.get(id=self.financialaid.id)
+        assert resp.status_code == status.HTTP_200_OK
+        assert financialaid.tier_program == self.tier_programs["50k"]
+        assert financialaid.status == FinancialAidStatus.APPROVED
+        assert mock_mailgun_client.send_individual_email.called
+        _, called_kwargs = mock_mailgun_client.send_individual_email.call_args
+        assert called_kwargs['subject'] == FINANCIAL_AID_APPROVAL_SUBJECT_TEXT
+        assert called_kwargs['body'] == FINANCIAL_AID_APPROVAL_MESSAGE_BODY
+        assert called_kwargs['recipient'] == self.profile.user.email
+
+    def test_financial_aid_action_view_with_rejection(self, mock_mailgun_client):
+        """
+        Tests FinancialAidActionView when application is rejected
+        """
+        mock_mailgun_client.send_individual_email.return_value = Mock(
+            spec=Response,
+            status_code=status.HTTP_200_OK,
+            json=mocked_json()
+        )
+        self.financial_review_data["action"] = FinancialAidStatus.REJECTED
+        self.client.force_login(self.staff_user_profile.user)
+        resp = self.client.post(self.action_url, data=self.financial_review_data)
+        assert resp.status_code == status.HTTP_200_OK
+        financialaid = FinancialAid.objects.get(id=self.financialaid.id)
+        assert financialaid.tier_program == self.tier_programs["100k"]
+        assert financialaid.status == FinancialAidStatus.REJECTED
+        assert mock_mailgun_client.send_individual_email.called
+        _, called_kwargs = mock_mailgun_client.send_individual_email.call_args
+        assert called_kwargs['subject'] == FINANCIAL_AID_REJECTION_SUBJECT_TEXT
+        assert called_kwargs['body'] == FINANCIAL_AID_REJECTION_MESSAGE_BODY
+        assert called_kwargs['recipient'] == self.profile.user.email
+
+    def test_financial_aid_action_view_with_invalid_data(self, *args):  # pylint: disable=unused-argument
+        """
+        Tests FinancialAidActionView when invalid data is posted
+        """
+        # Invalid action
+        self.financial_review_data["action"] = FinancialAidStatus.PENDING_DOCS
+        self.client.force_login(self.staff_user_profile.user)
+        resp = self.client.post(self.action_url, data=self.financial_review_data)
+        assert resp.status_code == status.HTTP_400_BAD_REQUEST
+        # Invalid tier
```

Comment:
Please split these up into individual tests. As individual tests it is easier to identify which cases are failing and to prevent one case from impacting another case.